### ISSUE TITLE
BooleanCrosses returns false for basic intersections

### ISSUE THREAD

**Issue by chriserickson** [Turfjs/turf#1469]
booleanCrosses for 2 line segments returns false for a simple "X".

Given 2 line segments:
[0,0], [2,2]
and
[2,0], [0, 2]
who intersect at [1,1],

booleanCrosses returns false.

I've been debugging the code, and don't understand why the code requres that an endpoint of second feature to lie along the line of the first segment.

```
function doLineStringsCross(lineString1, lineString2) {
    var doLinesIntersect = lineIntersect(lineString1, lineString2);
    if (doLinesIntersect.features.length > 0) {
        for (var i = 0; i < lineString1.coordinates.length - 1; i++) {
            for (var i2 = 0; i2 < lineString2.coordinates.length - 1; i2++) {
                var incEndVertices = true;
                if (i2 === 0 || i2 === lineString2.coordinates.length - 2) {
                    incEndVertices = false;
                }

/// WHY THIS???
                if (isPointOnLineSegment(lineString1.coordinates[i], lineString1.coordinates[i + 1], lineString2.coordinates[i2], incEndVertices)) {
                    return true;
                }

            }
        }
    }
    return false;
}
```

**Comment by chriserickson** [comment]
I'd be happy to propose a fix for this, but would like to understand the intent before I try and fix it. 

@rowanwins, this code has your name on it, do you have a second to enlighten me?

**Comment by chriserickson** [comment]
@rowanwins : It looks like it was there to detect a touching, but not crossing line. I fixed it and filed a PR (https://github.com/Turfjs/turf/pull/1472).

I did pose the question on the PR though:
What is the correct behavior for the case where one line lies along another, or overlaps the endpoint of another

Case 1
Line 1: [0,0], [5, 0]
Line 2: [1, 0], [2, 0]

Case 2
Line 1: [0,0], [5, 0]
Line 2: [3, 0], 7, 0]

In both the previous and current implementation, these return "false" (I think). Is that what you'd expect?

**Comment by rowanwins** [comment]
Hey @chriserickson 

Yeah that is ringing bells. From memory looking at the definition the they needed to have interior points touching, hence why it couldn't just be an endpoint. 
I find referring back to the PostGIS ([such as this](https://postgis.net/docs/ST_Crosses.html)) quite helpful.

Thanks for the fix

**Comment by chriserickson** [comment]
Well, for the collinear lines, they technically do have an infinite number of touching interior points.

PostGIS returns a "true" in this case:
```sql
select st_intersects(
ST_GeomFromGeoJSON('{"type":"LineString","coordinates":[[0,0],[0,5]]}'),
ST_GeomFromGeoJSON('{"type":"LineString","coordinates":[[0,1],[0,4]]}')
)

-- true
```

**Comment by chriserickson** [comment]
I'll file a separate issue for that one.

**Comment by mfedderly** [comment]
I believe this was fixed by #1472. Please reopen if that is incorrect.
